docs/source/miscellaneous/doc/adr/0011-retry-long-running-locks-via-blockingtimeoutexception.md

```diff
@@ -0,0 +1,161 @@
+# 11. Retry long-running locks via BlockingTimeoutException
+
+Date: 08/05/2017
+
+## Status
+
+Accepted
+
+## Context
+
+Our implementation of AtlasDB clients and the TimeLock server were interacting in ways that were causing the TimeLock
+server to experience large thread buildups when running with HTTP/2. This manifested in 
+[issue #1680](https://github.com/palantir/atlasdb/issues/1680). The issue was eventually root caused to long-running
+lock requests in excess of the Jetty idle timeout; the server would close the relevant HTTP/2 stream, but *not*
+free up the resources consumed by the request. Eventually, these would overwhelm the TimeLock server, resulting
```

Comment:
'overwhelm' seems vague, particularly when most of these threads are probably sleeping waiting on a lock.  'eventually all server threads would be busy handling lock requests that had timed out'?  'eventually all server threads would be stuck waiting but no client would be listening for a response'?